axelor-account/src/main/java/com/axelor/apps/account/service/config/AccountConfigService.java

```diff
@@ -644,4 +660,85 @@ public Partner getDasContactPartner(AccountConfig accountConfig) throws AxelorEx
 
     return accountConfig.getDasContactPartner();
   }
+
+  public void checkChangesInAnalytic(
+      List<AnalyticAxisByCompany> initialList, List<AnalyticAxisByCompany> modifiedList)
+      throws AxelorException {
+    if (checkChangesInAnalyticConfig(initialList, modifiedList)) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(IExceptionMessage.ACCOUNT_CONFIG_ANALYTIC_CHANGE_IN_CONFIG));
+    }
+  }
+
+  public boolean checkChangesInAnalyticConfig(
+      List<AnalyticAxisByCompany> initialList, List<AnalyticAxisByCompany> modifiedList)
+      throws AxelorException {
+    if (!CollectionUtils.isEmpty(initialList)) {
+      List<AnalyticAxis> analyticAxisUsedList = getUsedAnalyticAxis(initialList);
+      if (initialList.size() < modifiedList.size()
+          && !CollectionUtils.isEmpty(analyticAxisUsedList)) {
+        return true;
+      } else if (initialList.size() > modifiedList.size()
+          && !CollectionUtils.isEmpty(analyticAxisUsedList)) {
+        return true;
+      } else if (initialList.size() == modifiedList.size()) {
+        if (axisChangedInConfig(initialList, modifiedList) != null
+            && analyticAxisUsedList.contains(axisChangedInConfig(initialList, modifiedList))) {
+          return true;
+        } else if (orderChangedInConfig(initialList, modifiedList)
+            && !CollectionUtils.isEmpty(analyticAxisUsedList)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  public List<AnalyticAxis> getUsedAnalyticAxis(List<AnalyticAxisByCompany> initialList) {
+    List<AnalyticAxis> analyticAxisList = new ArrayList<AnalyticAxis>();
+    for (AnalyticAxisByCompany analyticAxisByCompany : initialList) {
+      if (!CollectionUtils.isEmpty(
+          analyticMoveLineRepository
+              .findByAnalyticAxis(analyticAxisByCompany.getAnalyticAxis())
```

Comment:
Please use `count()` instead of `fetch()` for these cases, it will be more optimized.